inference-engine/src/transformations/src/transformations/op_conversions/simplify_ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "itt.hpp"
+
+#include "transformations/op_conversions/simplify_ctc_greedy_decoder_seq_len.hpp"
+
+#include <ngraph/opsets/opset6.hpp>
+#include <ngraph/rt_info.hpp>
+
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::SimplifyCTCGreedyDecoderSeqLen, "SimplifyCTCGreedyDecoder", 0);
+
+ngraph::pass::SimplifyCTCGreedyDecoderSeqLen::SimplifyCTCGreedyDecoderSeqLen() {
+    MATCHER_SCOPE(SimplifyCTCGreedyDecoderSeqLen);
+    auto decoder = pattern::wrap_type<opset6::CTCGreedyDecoderSeqLen>();
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        auto decoder_seq_len = std::dynamic_pointer_cast<opset6::CTCGreedyDecoderSeqLen> (m.get_match_root());
+        if (!decoder_seq_len) {
+            return false;
+        }
+
+        if (decoder_seq_len->get_input_size() > 2) {
+            const auto seq_len_pshape = decoder_seq_len->get_input_partial_shape(0);
+            auto blank_index = std::dynamic_pointer_cast<ngraph::opset6::Constant>(decoder_seq_len->input_value(2).get_node_shared_ptr());
+            if (!blank_index || seq_len_pshape.rank().is_dynamic() || seq_len_pshape[2].is_dynamic()) {
+                return false;
+            }
+
+            const std::vector<int64_t> &blank_index_values = blank_index->cast_vector<int64_t>();
+            const auto num_classes = decoder_seq_len->get_input_partial_shape(0)[2].get_length();
```

Comment:
You need to check that partial shape rank is static and after that check that dimension is static and only aftter you can call get_lengh. So that is why I asked you to use pattern::has_static_dim(2) in pattern to avoid this checks inside transformation code.